File: main_project/ReplicaCluster/ReplicaClusterDemo/src/server/cxx/ReplicaClusterDemoI.cpp
#include "ReplicaClusterDemoI.h"
#include "ServerStateSubscriber.h"

using namespace MyUtil;
using namespace xce::serverstate;
using namespace xce::replicaclusterdemo;
	
void MyUtil::initialize() {
	ServiceI& service = ServiceI::instance();
	service.getAdapter()->add(new DemoManagerI, service.createIdentity("M", ""));


	Ice::PropertiesPtr prop = service.getCommunicator()->getProperties();
	string serverName = prop->getPropertyWithDefault("Service.Name", "");
	ServerStateSubscriber::instance().initialize("ControllerDemo", serverName, 40);
}

void SetServerName(const string& serverName) {
	_serverName = serverName;
}

StrSeq DemoManagerI::PrintClients(const Ice::Current& current)
{
    return _serverName;
}

File: main_project/util/java/OceHA/src/main/java/xce/replicaclusterdemo/ReplicaClusterDemoI.java
package xce.replicaclusterdemo;

import xce.serverstate.ServerStateSubscriber;
import Ice.Current;

public class ReplicaClusterDemoI extends _repcludemoManagerDisp{
	
	private ServerStateSubscriber serverState;
	
	public ReplicaClusterDemoI(String servicename){
		
		serverState=ServerStateSubscriber.getInstance("Repcludemo");
		
		serverState.initialize(servicename,"ControllerFootprint", 300);

	}
	
	public void printprx(String bprx, Current __current) {
		System.out.println("The prx is : "+bprx);
	}
}
